### ISSUE TITLE
Question: environment variables with `wrangler dev`?

### ISSUE THREAD

**Issue by dchambers** [cloudflare/wrangler#1773]
Not sure whether this is just me doing the wrong thing or an actual bug report at this point, but I'm seeing that environment variables aren't available locally when I run `wranger dev`, but they work fine for the same script published using `wrangler publish`. Is there some trick here to get this to work?

Some info in case it's relevant:

```
node -v && wrangler -V && cat wrangler.toml
v14.15.4
👷 ✨  wrangler 1.12.2
name = "service-layer"
type = "javascript"
account_id = "<redacted>"
workers_dev = true

[env.staging]
name = "staging"
vars = { CLOUDFLARE_SCHEDULER_API_KEY = "expected-key" }

[env.production]
name = "production"
vars = { CLOUDFLARE_SCHEDULER_API_KEY = "expected-key" }
```

**Comment by xtuc** [comment]
do you pass `--env staging/production` when running wrangler dev or publish?

**Comment by dchambers** [comment]
> -env staging

Yes, I'd been using `--env staging` for both `wrangler publish` and `wrangler dev`. I did wonder if I needed to have _dev_ specific config, so added some `vars` underneath a `[dev]` section like so:

```
[dev]
vars = { CLOUDFLARE_SCHEDULER_API_KEY = "scheduler-key" }
```

but I got this error informing me that this is not a a valid configuration:

```
Error: unknown field `vars`, expected one of `ip`, `port`, `local_protocol`, `upstream_protocol`
```

However, I just now tried adding the `vars` at the global level, and this actually worked! Thanks for channeling me in the right direction 😄. 

I guess a warning if `--env` or `-e` are used with `workers dev` might help others in the future, though maybe they'll just find this issue too.
